pkg/convert/tf.go

```diff
@@ -1231,10 +1231,32 @@ func convertTemplateExpr(state *convertState,
 			tokens = append(tokens, makeToken(hclsyntax.TokenTemplateSeqEnd, "}"))
 		}
 	}
-	tokens = append(tokens, makeToken(hclsyntax.TokenCQuote, "\""))
+	if isHereDoc {
+		tokens = append(tokens, makeToken(hclsyntax.TokenCHeredoc, fmt.Sprintf("\n%s", cDelim)))
+	} else {
+		tokens = append(tokens, makeToken(hclsyntax.TokenCQuote, "\""))
+	}
 	return tokens
 }
 
+// detectHeredocDelim detects if the range contains a heredoc token and returns
+// the open and close delimiters.
+func detectHeredocDelim(state *convertState, r hcl.Range) (string, string, bool) {
+	hereDocRegex := regexp.MustCompile(`^<<-?([[:alpha:]]+)\n`)
```

Comment:
I think we might need to do better than this. If there's a hyphen, we need to trim leading whitespace from each line (assuming we aren't doing this elsewhere). Also, since `hereDocRegex` is a constant, it might be worth hoisting out (also for consistency I guess it should be ` heredocRegex`?).

Notes on `<<-`:

> If a heredoc template is introduced with the <<- symbol, any literal string at the start of each line is analyzed to find the minimum number of leading spaces, and then that number of prefix spaces is removed from all line-leading literal strings. The final closing marker may also have an arbitrary number of spaces preceding it on its line.

(From https://github.com/hashicorp/hcl/blob/main/hclsyntax/spec.md)

Reply:
I don't think we need to trim the whitespace, the HCL parser will do that when the PCL is reparsed, won't it?  We are just regenerating the PCL compliant HCL, which is parsed again later with HCL utilities anyways.  No need for us to remove the leading whitespace.

Reply:
Aye, we just need to emit it in exactly the same way